### ISSUE TITLE
Ackley(150, 150) performance

### ISSUE THREAD

**Issue by jacktang** [XZT008/Standard-GP-is-all-you-need-for-HDBO#1]
Hello, 

First of all, thanks for the interesting paper, I've read the paper and try to run the code. But I found that I could never reach the performance on the Ackley(150, 150) problem like below:

<img width="718" alt="Image" src="https://github.com/user-attachments/assets/fa2e219f-642f-437d-95e8-2cc35cd82b3c" />

For SBO(Matern), I got -16.603 at 400 iterations (the image result is around -8), And the logs: 
```
Seed: 0 --- At itr: 387: best value before=-16.701991240300348, after=-16.674492087986422, current query: [-16.67449209], time used: 23.819716215133667
Seed: 0 --- At itr: 388: best value before=-16.674492087986422, after=-16.674492087986422, current query: [-16.77843471], time used: 24.80426597595215
Seed: 0 --- At itr: 389: best value before=-16.674492087986422, after=-16.60807183405259, current query: [-16.60807183], time used: 21.342859983444214
Seed: 0 --- At itr: 390: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.78797585], time used: 21.462921142578125
Seed: 0 --- At itr: 391: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.67249121], time used: 29.97733783721924
Seed: 0 --- At itr: 392: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.70075605], time used: 21.65613293647766
Seed: 0 --- At itr: 393: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.70887957], time used: 21.90939426422119
Seed: 0 --- At itr: 394: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.73699198], time used: 21.966087818145752
Seed: 0 --- At itr: 395: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.64302319], time used: 20.67548108100891
Seed: 0 --- At itr: 396: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.71195564], time used: 26.228291034698486
Seed: 0 --- At itr: 397: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.64262342], time used: 30.73542308807373
Seed: 0 --- At itr: 398: best value before=-16.60807183405259, after=-16.60807183405259, current query: [-16.7463614], time used: 23.828952074050903
Seed: 0 --- At itr: 399: best value before=-16.60807183405259, after=-16.60363278436249, current query: [-16.60363278], time used: 23.089804887771606
Seed: 0 --- At itr: 400: best value before=-16.60363278436249, after=-16.60363278436249, current query: [-16.745541], time used: 21.50089406967163
```

And I got -11.107 for VBO at iteration 400 (the image result is around -5), and the logs list below:

```
Seed: 0 --- At itr: 393: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.49675476]
ls mean: 1.2470266820139841, ls std: 0.1033999608177991, max: 1.5148878440746674, min: 0.839288759397521
Seed: 0 --- At itr: 394: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.11870308]
ls mean: 1.247237938848984, ls std: 0.10324604643351873, max: 1.5115504583280641, min: 0.8390563191131455
Seed: 0 --- At itr: 395: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.02200296]
ls mean: 1.2493892975655472, ls std: 0.103889115513779, max: 1.5137743196906932, min: 0.8383806194407137
Seed: 0 --- At itr: 396: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.36319805]
ls mean: 1.249976213082145, ls std: 0.10327869587886901, max: 1.5027863799699397, min: 0.8385216727355
Seed: 0 --- At itr: 397: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.57425552]
ls mean: 1.251171738580321, ls std: 0.10380804975294709, max: 1.499705336677207, min: 0.8384645021518523
Seed: 0 --- At itr: 398: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.82446779]
ls mean: 1.25238934906859, ls std: 0.10407769998524732, max: 1.4979958444715804, min: 0.8391278590930831
Seed: 0 --- At itr: 399: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.2701635]
ls mean: 1.254142784960905, ls std: 0.10441957748086979, max: 1.4993204375188904, min: 0.8384686630577677
Seed: 0 --- At itr: 400: best value before=-11.107497633703414, best value after=-11.107497633703414, current query: [-14.73519941]
```

Is there something I missed? and I tried the code on different machines and got the same results.

**Comment by XZT008** [comment]
Hi Jack, may I know that data seed you used? Ackley is the function that have big variance with different data seed. I just checked my results across 20 seeds(0-19) and the worst is -11.

I will run with your seed and paste the result here.

**Comment by jacktang** [comment]
> Ackley is the function that have big variance with different data seed.

I didn't noticed it before, and my seed is 0.

**Comment by XZT008** [comment]
Hi Jack, so I rerun SBO Matern across different CPU platforms. It seems there are some instability issues during mll optimization on certain CPU. I modified the code, changed the optimizer from ADAM to RMSPROP with stopping criterion and the training is stable now. I will paste my results under(I only rerun seed 0-9 for simplicity)
```
Seed: 0 --- At itr: 420: best value before=-5.549302141326258, best value after=-5.549302141326258, current query: [-6.01573317]
Seed: 1 --- At itr: 420: best value before=-8.597564159850986, best value after=-8.597564159850986, current query: [-9.15844675]
Seed: 2 --- At itr: 420: best value before=-7.311257582613473, best value after=-7.311257582613473, current query: [-7.34338336]
Seed: 3 --- At itr: 420: best value before=-5.703046128349108, best value after=-5.703046128349108, current query: [-6.23812518]
Seed: 4 --- At itr: 420: best value before=-6.419828385390257, best value after=-6.384982034368164, current query: [-6.38498203]
Seed: 5 --- At itr: 420: best value before=-5.635065776602431, best value after=-5.635065776602431, current query: [-5.80945071]
Seed: 6 --- At itr: 420: best value before=-5.376512310471796, best value after=-5.376512310471796, current query: [-5.56411444]
Seed: 7 --- At itr: 420: best value before=-10.595582620771523, best value after=-10.455764358286073, current query: [-10.45576436]
Seed: 8 --- At itr: 420: best value before=-6.021473691937267, best value after=-6.021473691937267, current query: [-6.32012189]
Seed: 9 --- At itr: 420: best value before=-5.586949457059681, best value after=-5.586949457059681, current query: [-6.23183693]
```
Now the mean is of -6.67, which is significantly better than what we reported in paper(-7.83). Please let me know if you could achieve similar result as mine.

**Comment by XZT008** [comment]
For VBO, I suggest you to modify the learning rate and optimizer if you fit mll with botorch.fit.fit_gpytorch_mll_torch(). Or you can use Adam with your own training loop. That VBO implementation was written by myself, before their authors release the code, so you can also check their repo.

**Comment by jacktang** [comment]
Hello @XZT008 , here is the result of SBO Matern my side:

```
Seed: 0 --- At itr: 400: best value before=-15.464386548612039, after=-15.464386548612039, current query: [-15.48205357], time used: 24.71078586578369
Seed: 0 --- At itr: 401: best value before=-15.464386548612039, after=-15.387133628055226, current query: [-15.38713363], time used: 32.4711799621582
Seed: 0 --- At itr: 402: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.61665432], time used: 23.13524603843689
Seed: 0 --- At itr: 403: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.44055204], time used: 23.5293025970459
Seed: 0 --- At itr: 404: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.45051888], time used: 25.334946155548096
Seed: 0 --- At itr: 405: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.5827624], time used: 26.614386081695557
Seed: 0 --- At itr: 406: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.39980757], time used: 23.105879068374634
Seed: 0 --- At itr: 407: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.53480807], time used: 25.093026876449585
Seed: 0 --- At itr: 408: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.45807591], time used: 24.19081687927246
Seed: 0 --- At itr: 409: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.45782866], time used: 24.572007179260254
Seed: 0 --- At itr: 410: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.52469225], time used: 25.705748081207275
Seed: 0 --- At itr: 411: best value before=-15.387133628055226, after=-15.387133628055226, current query: [-15.58355591], time used: 23.92549204826355
Seed: 0 --- At itr: 412: best value before=-15.387133628055226, after=-15.332061528406893, current query: [-15.33206153], time used: 24.486002922058105
Seed: 0 --- At itr: 413: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.45232308], time used: 24.55023503303528
Seed: 0 --- At itr: 414: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.42815473], time used: 25.042925119400024
Seed: 0 --- At itr: 415: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.40571635], time used: 25.34359908103943
Seed: 0 --- At itr: 416: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.43650765], time used: 24.93643593788147
Seed: 0 --- At itr: 417: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.46216325], time used: 26.17358112335205
Seed: 0 --- At itr: 418: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.77864868], time used: 26.250406980514526
Seed: 0 --- At itr: 419: best value before=-15.332061528406893, after=-15.332061528406893, current query: [-15.41273348], time used: 24.992111921310425
Seed: 0 --- At itr: 420: best value before=-15.332061528406893, after=-15.318974287146194, current query: [-15.31897429], time used: 26.161269187927246
```
And with the python libs: 

```
botorch                    0.14.0
gpytorch                   1.14
torch                      2.2.2
python                     3.11.8
```

**Comment by XZT008** [comment]
**Please use the exact setting as in setup.py and follow the installation as in README.**

I pasted my result for seed 0 here. This is run on Intel 13th gen, similar results are achieved with Intel Xeon Gold 6244 and  Intel Xeon Cascadelake.
```
Seed: 0 --- At itr: 21: best value before=-21.079277206611078, best value after=-21.079277206611078, current query: [-21.57031115]
Seed: 0 --- At itr: 50: best value before=-21.079277206611078, best value after=-21.079277206611078, current query: [-21.52813584]
Seed: 0 --- At itr: 100: best value before=-15.415331781504944, best value after=-15.415331781504944, current query: [-16.26590128]
Seed: 0 --- At itr: 150: best value before=-11.893712953207487, best value after=-11.893712953207487, current query: [-11.93676339]
Seed: 0 --- At itr: 200: best value before=-10.040340026151428, best value after=-10.040340026151428, current query: [-10.18664397]
Seed: 0 --- At itr: 250: best value before=-8.854120555898248, best value after=-8.854120555898248, current query: [-9.07371433]
Seed: 0 --- At itr: 300: best value before=-7.4137807585059665, best value after=-7.4137807585059665, current query: [-7.72985463]
Seed: 0 --- At itr: 350: best value before=-6.707948922550052, best value after=-6.707948922550052, current query: [-6.94031332]
Seed: 0 --- At itr: 400: best value before=-5.741025746371129, best value after=-5.741025746371129, current query: [-5.99627279]
Seed: 0 --- At itr: 420: best value before=-5.704385431726493, best value after=-5.704385431726493, current query: [-6.24241713]
```

**Comment by jacktang** [comment]
I re-created the conda env, and now it got the similar result 👍 . Thanks @XZT008 !
